### ISSUE TITLE
each method fails to unify

### ISSUE THREAD

**Issue by chadaustin** [andyfriesen/crux#36]
```
fun handleFoldersRequest(response, root) {
    let rv = [];
    let contents: Array String = fs.readdirSync(root);
    contents->each(fun(name) {
    });
}
```

This fails to unify for some reason.  I get:

```
crux: Unification error at 28,5
	String
	TQuant 21
```

The equivalent:

```
each(contents, fun(name) {
});
```

Works fine.

**Comment by chadaustin** [comment]
I wonder if the solution for this and #37 is to _first_ desugar EMethodApp into EApp and then typecheck it like normal.  This would require converting UnresolvedReference into a sum type: `UnknownReference Name`, `QualifiedReference ImportName Name`, and perhaps `KnownReference ResolvedReference`.

Maybe I'll take a stab at that on the train.

**Comment by chadaustin** [comment]
Progress started on tdnr-unification branch.

**Comment by chadaustin** [comment]
Fixed a different way with 507b82f.
